api/src/main/java/org/openmrs/module/auditlog/service/impl/AuditLogServiceImpl.java

```diff
@@ -21,14 +21,12 @@
 import java.util.UUID;
 import java.util.stream.Collectors;
 
-@Service
 @Transactional
 public class AuditLogServiceImpl implements AuditLogService {
 
     private AuditLogDao auditLogDao;
-    
-    @Autowired
-    public AuditLogServiceImpl(AuditLogDao auditLogDao) {
+
+    public void setAuditLogDao(AuditLogDao auditLogDao) {
```

Comment:
prefer constructor argument instead of setter DI

Reply:
Addressed the comment

Reply:
In general, constructor args, if this is mandatory for class to be setup. If optional, then setters. That's how i have come to understand it.